### ISSUE TITLE
Rethink ant_metrics for H3C/H4C

### ISSUE THREAD

**Issue by jsdillon** [HERA-Team/hera_qm#341]
I've been doing some work on #331 and I realized that with the redcorr metrics gone (#334) we no longer need to store full set of visibilities. In fact, if we look at how MeanVij is calculated:

https://github.com/HERA-Team/hera_qm/blob/bab3dff9ffc32adcf42015dbadb681e2c780a32f/hera_qm/ant_metrics.py#L135-L153

we can see that it's effectively assigning the sum of the absolute value of all visibilities associated with an antenna, divided by the count of finite visibilities. Assuming that non-finite visibilities are rare and/or equally distributed across all baselines, this is basically equivalent to just computing the average visibility magnitude per baseline, then combining those into antennas based on which antennas are considered not excluded.

So basically, we just need to load each visibility, reduce it it to one number, and then decide later whether it gets included in an antenna's metric based on whether the other antenna involved is also being considered.

This raises the question: what's the best way to reduce a visibility magnitude to a single number?

**Comment by jsdillon** [comment]
I've been playing with 2458838 and three different ways of doing this. One could do a 2D median, a 2D mean, or take a median along the frequency direction and then the mean along the time dimension.

I've been playing with these ideas in the context of both a "chunked" analysis (60 integrations at a time) and a "per-file" analysis (2 integrations at a time). I'm also playing with whether I want to set the threshold at 5 sigma (as it current is) or lower it to 4 sigma. Here's what I'm finding for the chunked analysis:

[chunked_Mean_Dead_Cut_4_Cross_Cut_4.pdf](https://github.com/HERA-Team/hera_qm/files/4909354/chunked_Mean_Dead_Cut_4_Cross_Cut_4.pdf)
[chunked_Mean_Dead_Cut_5_Cross_Cut_5.pdf](https://github.com/HERA-Team/hera_qm/files/4909355/chunked_Mean_Dead_Cut_5_Cross_Cut_5.pdf)
[chunked_Median_Dead_Cut_4_Cross_Cut_4.pdf](https://github.com/HERA-Team/hera_qm/files/4909356/chunked_Median_Dead_Cut_4_Cross_Cut_4.pdf)
[chunked_Median_Dead_Cut_5_Cross_Cut_5.pdf](https://github.com/HERA-Team/hera_qm/files/4909357/chunked_Median_Dead_Cut_5_Cross_Cut_5.pdf)

And here's what I've found for the per-file analysis:

[Mean_Dead_Cut_4_Cross_Cut_4.pdf](https://github.com/HERA-Team/hera_qm/files/4909358/Mean_Dead_Cut_4_Cross_Cut_4.pdf)
[Mean_Dead_Cut_5_Cross_Cut_5.pdf](https://github.com/HERA-Team/hera_qm/files/4909359/Mean_Dead_Cut_5_Cross_Cut_5.pdf)
[Median_Dead_Cut_4_Cross_Cut_4.pdf](https://github.com/HERA-Team/hera_qm/files/4909360/Median_Dead_Cut_4_Cross_Cut_4.pdf)
[Median_Dead_Cut_5_Cross_Cut_5.pdf](https://github.com/HERA-Team/hera_qm/files/4909361/Median_Dead_Cut_5_Cross_Cut_5.pdf)

I'm running the 2D median now for the chunked analysis (for 2 integrations, it shouldn't matter).

**Comment by jsdillon** [comment]
And here's the 2D median on the chunked analysis

[2D_Chunked_Median_Dead_Cut_5_Cross_Cut_5.pdf](https://github.com/HERA-Team/hera_qm/files/4909629/2D_Chunked_Median_Dead_Cut_5_Cross_Cut_5.pdf)
[2D_Chunked_Median_Dead_Cut_4_Cross_Cut_4.pdf](https://github.com/HERA-Team/hera_qm/files/4909630/2D_Chunked_Median_Dead_Cut_4_Cross_Cut_4.pdf)

**Comment by jsdillon** [comment]
In my experience calibrating 2458838, the actual "bad" antennas are 2, 11, 12, 14, 39, 85, and 120. 

The mean based analyses are picking up a lot of antennas that aren't actually bad--I worry that this that RFI is really skewing the mean and different antennas see the RFI different. So I'd recommend going with the median algorithm.

The rest of the choices don't make much of a difference. Without chunking and a 2D median, I think 4 sigma is too sensitive. So I'd recommend 5 sigma and 2D median as the default, but if one does chunking I can see dropping down to 4 sigma. This should probably be reassess when the array functionality is better understood and there are many more antennas.
